### PR TITLE
Fuzz test netebpfext helper(s)

### LINKED ISSUE #1134: ebpf_extension_user.c doesn't implement provider unload correctly
During provider unload, client's should be notified. This is not currently implemented, resulting in an inability to test some scenarios.

### PR DESCRIPTION
**Pull request by dthaler**
## Description

* Add fuzz testing of netebpfext helper functions (of which there is currently only one).
* Fix local VS build to install execution context fuzzer corpus
* Fix execution_context/libfuzz_harness.cpp to clean up program information providers.  It had two lists, would append to one and clear the other so the list got duplicate entries etc.
* Get rid of separate ebpf_extension_user.c and use the kernel implementation for both kernel and user mode now
* Combine two different NMR mock implementations

Addresses the helper function part of issue #1325 and also fixes #1134

## Testing

This includes tests changes.

## Documentation

No impact.

### CODE REVIEW

**Comment by Alan-Jowett** [comment]
Does this resolve #1134?

**Comment by dthaler** [comment]
> Does this resolve #1134?

As far as I know yes, and the PR description says so :)

**Review by Alan-Jowett — DISMISSED** [review]

**Comment by dthaler** [comment]
@shankarseal can you review/approve?

**Review by shankarseal — DISMISSED** [review]

File: ebpfapi/rpc_client.cpp [hunk 1]
```diff
@@ -4,6 +4,7 @@
 // Windows.h needs to be included before other headers.
 // It has a #define for WINAPI_FAMILY_PARTITION among others that control
 // the behavior of other Windows headers.
+#include <winsock2.h>
```
[the remark this answers is not in the document]
Comment (dthaler) [reply ebpfapi/rpc_client.cpp:7]: Not as far as I could tell.

**Comment by dthaler** [comment]
@Alan-Jowett @shankar rebased to fix merge conflict, please re-approve

File: scripts/create_execution_context_corpus.bat [hunk 2]
```diff
@@ -0,0 +1,9 @@
+@echo off
```
[the remark this answers is not in the document]
Comment (dthaler) [reply scripts/create_execution_context_corpus.bat:1]: I had problems getting powershell to work from a VS command.  Feel free to submit a pull request if you can get it to work.

**Review by Alan-Jowett — APPROVED** [review]

**Review by gtrevi — APPROVED** [review]

**Review by delaramamiri — APPROVED** [review]

**Comment by dthaler** [comment]
@Alan-Jowett @shankarseal rebased to fix merge conflict, please re-approve

**Comment by dthaler** [comment]
@shankarseal @Alan-Jowett more merge conflicts since other things got merged first, please re-approve

**Review by Alan-Jowett — APPROVED** [review]

**Review by shankarseal — APPROVED** [review]

**Pull request merged**